core/src/test/scala/doric/Equalities.scala

```diff
@@ -29,6 +29,17 @@ object Equalities {
         case _ => a === b
       }
 
+  implicit def eqMap[K: Equality, V: Equality]: Equality[Map[K, V]] =
+    (a: Map[K, V], b: Any) =>
+      b match {
+        case m: Map[_, _] =>
+          // a == m
+          if (a.keys === m.keys) {
+            a.values === m.values
+          } else false
+        case _ => a === b
+      }
```

Comment:
```suggestion
    case (a: Map[K, V], b: Map[K, V]) =>
          (a.keys === b.keys) && (a.values === b.values)
    case _ => false
      }
```

1. Less code and more focused in the only true possible case
2. Return a false in case an if is not true? delete the if in that case
3. case _ => I don't think that if `b `is not a map would be true in any case, also it will do a recursive call with the same type class.

Reply:
Change the rest of equality with the same problem